src/actions/question.js

```diff
@@ -158,18 +160,51 @@ const finishAnsweringQuestionFailure = makeActionCreator(
   'feedback'
 )
 
-export function finishAnsweringQuestion(queueId, questionId, feedback) {
+export function finishAnsweringQuestion(
+  queueId,
+  questionId,
+  shouldCheckFeedback,
+  feedback
+) {
   return dispatch => {
-    dispatch(finishAnsweringQuestionRequest(queueId, questionId, feedback))
-
-    return axios.post(`/api/questions/${questionId}/answered`, feedback).then(
-      () =>
-        dispatch(finishAnsweringQuestionSuccess(queueId, questionId, feedback)),
-      err => {
-        console.error(err)
-        dispatch(finishAnsweringQuestionFailure(queueId, questionId))
-      }
+    dispatch(
+      finishAnsweringQuestionRequest(
+        queueId,
+        questionId,
+        shouldCheckFeedback,
+        feedback
+      )
     )
+
+    if (shouldCheckFeedback) {
+      return axios
+        .post(`/api/questions/${questionId}/answeredWith`, feedback)
```

Comment:
Why two separate routes? Whether or not feedback is needed should be enforced by the server in a single route handler.